src/hooks/useApiThrottle.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+type ApiCall = (...args: any[]) => Promise<any>;
+
+interface UseApiThrottleProps<T extends ApiCall> {
+  fn: T;
+  callback?: (res: Awaited<ReturnType<T>>) => void;
+}
+
+/**
+ * Prevent additional API call until the most recent one has completed.
+ */
+const useApiThrottle = <T extends ApiCall>(props: UseApiThrottleProps<T>) => {
+  const [fetching, setFetching] = React.useState(false);
```

Comment:
could also `useRef` here in case the callback gets called twice before state updates

Reply:
Is `useRef` safe we use `current` to display the loading indicator?